src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/ClassMemberImpliedModifierCheck.java

```diff
@@ -243,7 +243,8 @@ public void visitToken(DetailAST ast) {
      * @return true if ast is in a class, enum, or record
      */
     private static boolean isInTypeBlock(DetailAST ast) {
-        return ScopeUtil.isInClassBlock(ast)
+        return ScopeUtil.isInObjBlock(ast)
+                || ScopeUtil.isInClassBlock(ast)
                 || ScopeUtil.isInEnumBlock(ast)
                 || ScopeUtil.isInRecordBlock(ast);
```

Comment:
All of these structures have an `OBJ_BLOCK` token; can we just check if `ast` is in an `OBJECT_BLOCK` only?

Reply:
No, we need to make sure it is not directly included in a Interface block

Reply:
Ah, yes. This is taken care of by https://checkstyle.sourceforge.io/config_modifier.html#InterfaceMemberImpliedModifier, thanks for the reminder.